### PR TITLE
Adding better sorting and organization of events

### LINKED ISSUE #9: Expired Events
The new events page displays expired events alongside upcoming ones. 

I don't know what the right approach is (a separate past events page, or a toggle switch to show old events, etc) but I don't think they should, by default, be displayed on the main events page. An easy way to search/display archived events is important and we want to capture it.

### LINKED ISSUE #34: Updates to events/navigation
- What is an RSE should be moved to about
- Add link to available jobs on front page (maybe replace resources)
-  Should we rename top menu item (careers) to something else? Job Board? need to imply it's not US-RSE careers, but job board.
- Category view of events should be sorted under categories.
 - Events archive should show older events (and archived events should be removed from category / upcoming)

### PR DESCRIPTION
**Pull request by vsoch**
This change renames careers to job board, adds a sorting to events in each category (only showing the top 5), a link below each category to show more, an events.yaml file to put a better descriptor for the event (so we do not use the tag), an event archive page organized by year, an event category page organized by category including the entire list, adding the job board to the front page, updating the second box on the front page to link to what is an rse, and reorganizing pages so that those under about are physically in that folder and the url has the about/ prefix

Or in simpler terms, I think this covers what we discussed today. It will close #9 and fix #34.

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

### CODE REVIEW

**Comment by mtbc** [comment]
If I click something like "View more dei events" then I am taken to a page that also has more community calls, etc. Is that expected? If so, I can see why, just wanted to make sure it's okay.

The "what is an RSE" from the front page takes me to http://127.0.0.1:4000/about/what-is-an-rse/ which fails because it really wants http://127.0.0.1:4000/usrse.github.io/about/what-is-an-rse/. Is there a deployment trick I should be using to better facilitate such local testing? I'm just doing the `bundle install`, `jekyll serve`, not seeing an Artifacts tab anywhere.

**Comment by vsoch** [comment]
@mtbc yes - right now it just goes to an expanded pages of events, and if I remember it should be at the section you clicked.

> The "what is an RSE" from the front page takes me to http://127.0.0.1:4000/about/what-is-an-rse/ which fails because it really wants http://127.0.0.1:4000/usrse.github.io/about/what-is-an-rse/. Is there a deployment trick I should be using to better facilitate such local testing? I'm just doing the bundle install, jekyll serve, not seeing an Artifacts tab anywhere.

There should be a `{{ site.baseurl }}` that grabs that url.

File: pages/index.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ full_width: true
        <h2 class="block-title">Great ideas deserve great software.</h2>
        <p>Research Software Engineers use expertise in programming to advance research.
       This includes researchers who spend a significant amount of time programming, full-time software engineers writing code to solve research problems, and those somewhere in-between. We aspire to apply the skills and practices of software development to research to create more robust, manageable, and sustainable research software.</p>
-    <a href="/docs/getting-started" class="button btn-warning">Read More</a>
+    <a href="/about/what-is-an-rse/" class="button btn-warning">Read More</a>
```
Comment (vsoch) [inline pages/index.md:33]: This should be the fix for the url.
```suggestion
    <a href="{{ site.baseurl }}/about/what-is-an-rse/" class="button btn-warning">Read More</a>
```

**Comment by mtbc** [comment]
This PR also looks to assist #18.

**Comment by mtbc** [comment]
> "View more dei events"

I suppose "View more DEI events" may be rather better. If so, would it be best to just leave it as `dei` under the hood and define a "short name" or somesuch for  the categories in `events.yaml`?

**Comment by vsoch** [comment]
I would let the details about the presentation of DEI (and pages when they exist) be under the decision of the DEI working group.

**Comment by mtbc** [comment]
:+1: The other category labels do fit more comfortably on their "view more" buttons.

**Comment by vsoch** [comment]
@mtbc I think I was confusing the other PR for this one - sorry about that on my part as well! Is this okay to merge? For the DEI details we can present that to the DEI group (of which I'm a part, so I suspect I'll bring this up at some point :laughing: )
